client/src/models/history/Robot6Axis_FC.ts

```diff
@@ -68,47 +67,15 @@ export const update: Update = async (apiType, productId, params) => {
     typeof updatedParamIndex === 'undefined' || param.index === updatedParamIndex
 
   // Update axis
-  if (check(paramsMap[a1])) {
-    await updateAxis(params.values, a1, api)
-  }
-
-  if (check(paramsMap[a2])) {
-    await updateAxis(params.values, a2, api)
-  }
-
-  if (check(paramsMap[a3])) {
-    await updateAxis(params.values, a3, api)
-  }
-
-  if (check(paramsMap[a4])) {
-    await updateAxis(params.values, a4, api)
-  }
-
-  if (check(paramsMap[a5])) {
-    await updateAxis(params.values, a5, api)
-  }
-
-  if (check(paramsMap[a6])) {
-    await updateAxis(params.values, a6, api)
+  for (let index = 0; index < 6; index++) {
+    if (check(paramsMap[index])) {
+      await api.updateFastenedConstraints({ ...constraints[index], zRotation: (params.values[index] / 180) * Math.PI })
+    }
```

Comment:
Just a note. If several parameters can be changed at the same time, the API call per parameter leads to performance losses. However, as the user interface only allows a single parameter to be changed, this is irrelevant for this application.

Reply:
You're completely right. In another application where we could change multiple parameters at once, we would have to change this.